File: Basics/Conditional Statements Advanced/Exercise/08.py
hour_exam = int(input())
minute_exam = int(input())
hour_arrival= int(input())
minute_arrival= int(input())

all_min_exam = (hour_exam * 60) + minute_exam
all_min_arrival = (hour_arrival * 60) + minute_arrival
diff_minutes = abs(all_min_arrival - all_min_exam)

if all_min_arrival > all_min_exam:
    print("Late")
    if diff_minutes < 60:
        minutes = diff_minutes % 60
        print(f"{minutes} minutes after the start")
    elif diff_minutes >= 60:
        hour = diff_minutes // 60
        minutes = diff_minutes % 60
        print(f"{hour}:{minutes:02d} hours after the start")

elif all_min_arrival == all_min_exam or diff_minutes <= 30:
    print("On time")
    if diff_minutes <= 30 and diff_minutes != 0:
        print(f"{diff_minutes} minutes before the start")

else:
    print("Early")
    if diff_minutes >= 60:
        hour = diff_minutes // 60
        minutes = diff_minutes % 60
        print(f"{hour}:{minutes:02d} hours before the start")
    else:
        print(f"{diff_minutes} minutes before the start")

File: JS-Basics/Conditional Statements Advanced/Exercise/08.js
function examArrival(hourExam, minuteExam, hourArrival, minuteArrival) {
    const examTime = new Date(0, 0, 0, hourExam, minuteExam);
    const arrivalTime = new Date(0, 0, 0, hourArrival, minuteArrival);

    const diffMilliseconds = arrivalTime - examTime;
    const diffMinutes = Math.abs(Math.floor(diffMilliseconds / (1000 * 60)));

    if (diffMilliseconds > 0) {
        console.log("Late");
        if (diffMinutes < 60) {
            console.log(`${diffMinutes} minutes after the start`);
        } else {
            const hours = Math.floor(diffMinutes / 60);
            const minutes = diffMinutes % 60;
            console.log(`${hours}:${minutes.toString().padStart(2, '0')} hours after the start`);
        }
    } else if (diffMilliseconds === 0 || diffMinutes <= 30) {
        console.log("On time");
        if (diffMinutes !== 0) {
            console.log(`${diffMinutes} minutes before the start`);
        }
    } else {
        console.log("Early");
        if (diffMinutes >= 60) {
            const hours = Math.floor(diffMinutes / 60);
            const minutes = diffMinutes % 60;
            console.log(`${hours}:${minutes.toString().padStart(2, '0')} hours before the start`);
        } else {
            console.log(`${diffMinutes} minutes before the start`);
        }
    }
}
